ospf6d/ospf6_nssa.c

```diff
@@ -307,6 +307,7 @@ void ospf6_abr_remove_unapproved_summaries(struct ospf6 *ospf6)
 			if (CHECK_FLAG(lsa->flag, OSPF6_LSA_UNAPPROVED)) {
 				lsa->header->age = htons(OSPF_LSA_MAXAGE);
 				THREAD_OFF(lsa->refresh);
+				THREAD_OFF(lsa->expire);
```

Comment:
Wouldn't it be better to call `ospf6_lsa_premature_aging()` here instead?

Reply:
dunno.  I just know that we are calling thread_execute() which executes *now*.  I just wanted to ensure that the expire thread is cleared, else we will have a thread pointer in lsa->expire that has a backpointer to a possibly freed lsa.